src/daft-core/src/array/ops/from_arrow.rs

```diff
@@ -145,3 +128,85 @@ impl FromArrow for StructArray {
         }
     }
 }
+
+impl FromArrow for MapArray {
+    fn from_arrow(field: FieldRef, arrow_arr: Box<dyn arrow2::array::Array>) -> DaftResult<Self> {
+        // we need to handle map type separately because the physical type of map in arrow2 is map but in Daft is list
+
+        println!("map");
+        match (&field.dtype, arrow_arr.data_type()) {
+            (DataType::Map { key, value }, arrow2::datatypes::DataType::Map(map_field, _)) => {
+                let arrow_arr = arrow_arr
+                    .as_any()
+                    .downcast_ref::<arrow2::array::MapArray>()
+                    .unwrap();
+                let arrow_child_array = arrow_arr.field();
+
+                let child_field = Field::new(
+                    map_field.name.clone(),
+                    DataType::Struct(vec![
+                        Field::new("key", *key.clone()),
+                        Field::new("value", *value.clone()),
+                    ]),
+                );
+                let physical_field = Field::new(
+                    field.name.clone(),
+                    DataType::List(Box::new(child_field.dtype.clone())),
+                );
+
+                let child_series =
+                    Series::from_arrow(child_field.into(), arrow_child_array.clone())?;
+                println!("start");
+
+                let physical = ListArray::new(
+                    physical_field,
+                    child_series,
+                    arrow_arr.offsets().into(),
+                    arrow_arr.validity().cloned(),
+                );
+                println!("end");
+
+                Ok(Self::new(field, physical))
+            }
+            (d, a) => Err(DaftError::TypeError(format!(
+                "Attempting to create Daft MapArray with type {} from arrow array with type {:?}",
+                d, a
+            ))),
+        }
+    }
+}
+
+macro_rules! impl_logical_from_arrow {
+    ($logical_type:ident) => {
+        impl FromArrow for LogicalArray<$logical_type> {
+            fn from_arrow(
+                field: FieldRef,
+                arrow_arr: Box<dyn arrow2::array::Array>,
+            ) -> DaftResult<Self> {
+                let target_convert = field.to_physical();
+                let target_convert_arrow = target_convert.dtype.to_arrow()?;
+
+                let physical_arrow_array = arrow_arr.convert_logical_type(target_convert_arrow);
```

Comment:
we can't just perform a `convert_logical_type` as this wont work for things like `largeutf8` -> `utf8`. We need a physical cast here.

Reply:
This should actually be fine since the only types that implement this are not nested types, and I believe have straightforward conversions for their underlying types in Arrow. The only problematic one is map which I have implemented differently.